### ISSUE TITLE
Does not respect JSON style YAML.

### ISSUE THREAD

**Issue by envygeeks** [atom/language-yaml#5]
Even though the spec doesn't explicitly display JSON style YAML it allows for it because of the massive spec and the way it allows certain kinds of sub-keys, however Atom's YAML parser does not recognize or respect keys and values in the following format (the way that it does with JSON itself):

``` yaml
{
  "key": "value"
}
```

**Comment by szhu** [comment]
Screenshot for illustration

![screen shot 2015-07-06 at 4 08 09 pm](https://cloud.githubusercontent.com/assets/1570168/8535312/81797e46-23f9-11e5-9a3a-ae6f0319ed68.png)

**Comment by franklinyu** [comment]
I think the YAML specification *do* explicitly display this. It's called [Flow Collection Styles](http://www.yaml.org/spec/1.2/spec.html#id2790088). Note that quotation mark is optional when key or value don't have whitespace:

```yaml
outerkey: {key: value, "key with whitespace": "value with whitespace"}
```

which is equivalent to

```yaml
outerkey:
  key: value
  "key with whitespace": value with whitespace
```

**Comment by 50Wliu** [comment]
@franklinyu Yep, I started working on it but never got the chance to finish it.

**Comment by jjlorenzo** [comment]
Hi any news from this?

**Comment by Nixinova** [comment]
Here's an entire file that is wrongly colored due to this bug: https://github.com/Nixinova/Minecraft-Versions/blob/HEAD/src/versions.yml
It thinks the first colon from a json key makes the entire preceding text the key
